atl-checker/src/algorithms/global/mod.rs

```diff
@@ -0,0 +1,466 @@
+                    }
+                }
+            }
+        }
+        changed_flag
+    }
+
+    /// Rising the value of the source vertex assignment if all targets are true or empty. If the
+    /// source vertex already is true we simply return. The return value is based on if a changed
+    /// have been made.
+    fn process_hyper(&mut self, edge: HyperEdge<V>) -> bool {
+        if *self.assignment.get(&edge.source).unwrap() {
+            false
+        } else {
+            let mut final_ass = true;
+            for target in edge.targets {
+                if !self.assignment.get(&target).unwrap() {
+                    final_ass = false;
+                    break;
+                }
+            }
+            self.update_assignment(edge.source, final_ass)
+        }
+    }
+
+    /// Rising the value of the source vertex assignment if the target vertex was assigned
+    /// false in the last component assignment. If the source vertex already is true we
+    /// simply return. The return value is based on if a changed have been made.
+    fn process_negation(
+        &mut self,
+        edge: NegationEdge<V>,
+        ass_from_earlier: HashMap<V, bool>,
+    ) -> bool {
+        if *self.assignment.get(&edge.source).unwrap() {
+            false
+        } else {
+            let mut final_ass = true;
+
+            if *ass_from_earlier.get(&edge.target).unwrap() {
+                final_ass = false;
+            }
+
+            self.update_assignment(edge.source, final_ass)
+        }
+    }
+
+    /// Updating an assignment to the new_ass value, if the new_ass are
+    /// equal to the old assignment, we simply return. The return value
+    /// is based on whether the assignment of the given vertex is changed or not.
+    fn update_assignment(&mut self, v: V, new_ass: bool) -> bool {
+        self.assignment
+            .get_mut(&v)
+            .map(|ass| {
+                if new_ass == *ass {
+                    false
+                } else {
+                    *ass = new_ass;
+                    true
+                }
+            })
+            .unwrap()
+    }
+
+    /// From the dist list the components are identified. Since the index
+    /// of every element in `self.dist` are representing the actually distance,
+    /// counted from the end of the list, we can simply pop the back of the list
+    /// until it is empty. For every pop we add the element to our components and
+    /// save them. The components list returned then have C_0 as it first element
+    /// followed by C_1 .. C_k.
+    fn get_components(&mut self) -> VecDeque<HashSet<V>> {
+        let mut components = VecDeque::<HashSet<V>>::new();
+        let mut component = HashSet::<V>::new();
+
+        while !self.dist.is_empty() {
+            for v in self.dist.pop_back().unwrap() {
+                component.insert(v);
+            }
+            components.push_back(component.clone());
+        }
+        components
```

Comment:
Looks like my biggest comment got yeeted... Here we go again:
If I read this correctly, component C_0 is a superset of C_1 which is a superset of C_2 etc? This matches the math, but it is actually unnecessary. Once the assignments of configurations of C_k has been found, they are never going to change. So if we want better performance, the components can be mutually exclusive.